### PR TITLE
Update Taquito dependencies to 15.0.1 and use ghostnet as testnet references

### PR DESCRIPTION
**Pull request by danielelisi**
Updating Taquito dependencies to `v15.0.1` and using Ghostnet as a reference in the config files. This is to avoid us having to update name for every new testnets

### CODE REVIEW

File: accounts-config.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "taquito-example": {
-    "kathmandu": {
-      "regular": "taquito-example-kathmandunet",
-      "ephemeral": "ephemeral-keys-kathmandunet"
+    "ghostnet": {
```
Comment (dsawali) [inline accounts-config.json:3]: would this also work for limanet? or do we have a different config file for that? not sure how the configs are structured

File: tsconfig.json [hunk 2]
```diff
@@ -1,13 +1,11 @@
 {
   "compilerOptions": {
+    "esModuleInterop": true,
```
Comment (dsawali) [inline tsconfig.json:3]: what would this enable?

File: accounts-config.json (hunk 1 above)
Comment (danielelisi) [reply accounts-config.json:3]: These are only example configs. It isn't what's deployed in our infra

File: tsconfig.json (hunk 2 above)
Comment (danielelisi) [reply tsconfig.json:3]: https://www.typescriptlang.org/tsconfig#esModuleInterop

I was getting the following error when building the docker image for keygen
```
#19 10.68 node_modules/@taquito/signer/dist/types/derivation-tools/ecdsa.d.ts(3,8): error TS1259: Module '"/node_modules/@types/bn.js/index"' can only be default-imported using the 'esModuleInterop' flag
#19 ERROR: process "/bin/sh -c npm run build" did not complete successfully: exit code: 2

#16 [linux/arm64 5/8] RUN npm ci
#16 CANCELED
------
 > [linux/amd64 8/8] RUN npm run build:
#19 0.576 
#19 0.576 > key-gen-api@1.0.0 build
#19 0.576 > tsc
#19 0.576 
#19 10.68 node_modules/@taquito/signer/dist/types/derivation-tools/ecdsa.d.ts(3,8): error TS1259: Module '"/node_modules/@types/bn.js/index"' can only be default-imported using the 'esModuleInterop' flag
------
Dockerfile:12
--------------------
  10 |     COPY src src
  11 |     
  12 | >>> RUN npm run build
  13 |     
  14 |     CMD node ./dist/lib/index.js
--------------------
ERROR: failed to solve: process "/bin/sh -c npm run build" did not complete successfully: exit code: 2
Error: buildx failed with: ERROR: failed to solve: process "/bin/sh -c npm run build" did not complete successfully: exit code: 2
```

**Review by dsawali — APPROVED** [review]

**Pull request merged**